pkg/manager/daemon_adaptor.go

```diff
@@ -25,6 +25,28 @@ import (
 	metrics "github.com/containerd/nydus-snapshotter/pkg/metrics/tool"
 )
 
+func (m *Manager) IsSubscribedDaemon(id, path string) bool {
+	return m.monitor.IsSubscribed(id, path)
+}
+
+func (m *Manager) StartDaemonUntilSubscribed(d *daemon.Daemon) {
+	for {
+		if err := m.StartDaemon(d); err != nil {
+			log.L.Errorf("fail to start daemon %s when recovering", d.ID())
+		}
+
+		running, err := d.IsProcessRunning()
+		if err != nil {
+			log.L.Errorf("fail to get process state, pid %d", d.Pid())
```

Comment:
Better to throw the `err.Error()` message together.

Reply:
Fixed.